### PR TITLE
Extract filtering cases on ticket

### PR DESCRIPTION
**Pull request by seallard**
### Fixed
- Extract filtering of cases on ticket

## Review
- [ ] Tests executed by
- [ ] "Merge and deploy" approved by

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/store/api/find_business_data.py [hunk 1]
```diff
@@ -198,17 +198,27 @@ def filter_cases_with_samples(self, case_ids: List[str]) -> List[str]:
                 cases_with_samples.add(case_id)
         return list(cases_with_samples)
 
-    def get_cases_from_ticket(self, ticket: str) -> Query:
-        return self.Family.query.filter(Family.tickets.contains(ticket))
+    def get_cases_by_ticket_id(self, ticket_id: str) -> List[Family]:
+        return apply_case_filter(
```
Comment (henrikstranneheim) [inline cg/store/api/find_business_data.py:202]: Add doc str

File: cg/store/api/find_business_data.py [hunk 2]
```diff
@@ -198,17 +198,27 @@ def filter_cases_with_samples(self, case_ids: List[str]) -> List[str]:
...
-            self.Family.query.filter(Family.tickets.contains(ticket)).first().customer.internal_id
-        )
+        cases: List[Family] = self.get_cases_by_ticket_id(ticket_id=ticket)
+        if not cases:
+            raise ValueError(f"No case found for ticket {ticket}")
+        return cases.customer.internal_id
 
     def get_samples_from_ticket(self, ticket: str) -> List[Sample]:
-        return self._get_join_sample_family_query().filter(Family.tickets.contains(ticket)).all()
+        """Returns the samples related to given ticket"""
```
Comment (henrikstranneheim) [inline cg/store/api/find_business_data.py:216]:
```suggestion
        """Returns the samples related to given ticket."""
```

File: cg/store/filters/status_case_filters.py [hunk 3]
```diff
@@ -111,18 +111,29 @@ def filter_case_by_internal_id(cases: Query, internal_id: str, **kwargs) -> Quer
     return cases.filter(Family.internal_id == internal_id)
 
 
+def filter_cases_by_ticket(cases: Query, ticket_id: str, **kwargs) -> Query:
```
Comment (henrikstranneheim) [inline cg/store/filters/status_case_filters.py:114]:
```suggestion
def filter_cases_by_ticket_id(cases: Query, ticket_id: str, **kwargs) -> Query:
```

File: tests/store/filters/test_status_cases_filters.py [hunk 4]
```diff
@@ -623,3 +624,31 @@ def test_filter_case_by_empty_internal_id(store_with_multiple_cases_and_samples:
 
     # THEN the query should return no cases
     assert cases.count() == 0
+
+
+def test_filter_cases_by_ticket_no_matching_ticket(
+    store_with_multiple_cases_and_samples: Store, non_existent_id: str
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_cases_filters.py:630]: Add doc str

File: tests/store/filters/test_status_cases_filters.py [hunk 5]
```diff
@@ -623,3 +624,31 @@ def test_filter_case_by_empty_internal_id(store_with_multiple_cases_and_samples:
...
+
+    # WHEN filtering cases by a non-existent ticket
+    filtered_cases: Query = filter_cases_by_ticket(cases=cases_query, ticket_id=non_existent_id)
+
+    # THEN the query should return no cases
+    assert filtered_cases.count() == 0
+
+
+def test_filter_cases_by_ticket_matching_ticket(
+    store_with_multiple_cases_and_samples: Store, ticket: str
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_cases_filters.py:643]: Add doc str
Is the fixture ticket actually ticket id?

File: tests/store/filters/test_status_cases_filters.py [hunk 6]
```diff
@@ -623,3 +624,31 @@ def test_filter_case_by_empty_internal_id(store_with_multiple_cases_and_samples:
...
+):
+    # GIVEN a store containing cases with a matching ticket
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_cases_filters.py:645]:
```suggestion
    # GIVEN a store containing cases with a matching ticket id
```

File: tests/store/filters/test_status_cases_filters.py [hunk 7]
```diff
@@ -623,3 +624,31 @@ def test_filter_case_by_empty_internal_id(store_with_multiple_cases_and_samples:
 
     # THEN the query should return no cases
     assert cases.count() == 0
+
+
+def test_filter_cases_by_ticket_no_matching_ticket(
+    store_with_multiple_cases_and_samples: Store, non_existent_id: str
+):
+    # GIVEN a store containing cases with no matching ticket
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_cases_filters.py:632]:
```suggestion
    # GIVEN a store containing cases with no matching ticket id
```

File: tests/store/filters/test_status_cases_filters.py [hunk 8]
```diff
@@ -623,3 +624,31 @@ def test_filter_case_by_empty_internal_id(store_with_multiple_cases_and_samples:
...
+    assert filtered_cases.count() == 0
+
+
+def test_filter_cases_by_ticket_matching_ticket(
+    store_with_multiple_cases_and_samples: Store, ticket: str
+):
+    # GIVEN a store containing cases with a matching ticket
+    cases_query: Query = store_with_multiple_cases_and_samples._get_query(table=Family)
+
+    # WHEN filtering cases by an existing ticket
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_cases_filters.py:648]:
```suggestion
    # WHEN filtering cases by an existing ticket id
```

**Review by henrikstranneheim — APPROVED** [review]
👍

File: tests/store/filters/test_status_cases_filters.py (hunk 5 above)
Comment (seallard) [reply tests/store/filters/test_status_cases_filters.py:643]: Yes, the fixture could have a better name. Changing it

**Comment by seallard** [comment]
deployed to stage and prod

**Pull request merged**